core/server/src/main/java/alluxio/worker/block/BlockWorkerClientRestServiceHandler.java

```diff
@@ -239,46 +243,40 @@ public Response promoteBlock(@QueryParam("blockId") Long blockId) {
   @Path(READ_BLOCK)
   @Produces(MediaType.APPLICATION_OCTET_STREAM)
   @ReturnType("java.util.List<java.lang.Byte>")
-  public Response readBlock(@QueryParam("sessionId") Long sessionId,
-      @QueryParam("blockId") Long blockId, @QueryParam("lockId") Long lockId,
-      @QueryParam("offset") Long offset, @QueryParam("length") Long length) {
-    try {
-      Preconditions.checkNotNull(blockId, "required 'blockId' parameter is missing");
-      Preconditions.checkNotNull(sessionId, "required 'sessionId' parameter is missing");
-      Preconditions.checkNotNull(lockId, "required 'lockId' parameter is missing");
-      Preconditions.checkNotNull(offset, "required 'offset' parameter is missing");
-      Preconditions.checkNotNull(length, "required 'length' parameter is missing");
-      Preconditions.checkState(offset >= 0, "invalid offset: %s", offset);
-      Preconditions.checkState(length >= -1, "invalid length (except for -1): %s", length);
-      // TODO(jiri): Wrap this logic in a block worker function; requires refactoring.
-      BlockReader reader = null;
-      try {
-        reader = mBlockWorker.readBlockRemote(sessionId, blockId, lockId);
-        long fileLength = reader.getLength();
-        Preconditions
-            .checkArgument(offset <= fileLength, "offset %s is larger than file length %s", offset,
-                fileLength);
-        Preconditions.checkArgument(length == -1 || offset + length <= fileLength,
-            "offset %s plus length %s is larger than file length %s", offset, length, fileLength);
-        long readLength = (length == -1) ? fileLength - offset : length;
-        ByteBuffer buffer = reader.read(offset, readLength);
-        mBlockWorker.accessBlock(sessionId, blockId);
-        if (buffer.hasArray()) {
-          return RestUtils.createResponse(buffer.array());
-        }
-        // We need to copy the bytes because the buffer byte array cannot be accessed directly.
-        byte[] bytes = new byte[(int) readLength];
-        buffer.get(bytes);
-        return RestUtils.createResponse(bytes);
-      } finally {
-        if (reader != null) {
-          reader.close();
+  public Response readBlock(@QueryParam("sessionId") final Long sessionId,
+      @QueryParam("blockId") final Long blockId, @QueryParam("lockId") final Long lockId,
+      @QueryParam("offset") final Long offset, @QueryParam("length") final Long length) {
+    return RestUtils.call(new RestUtils.RestCallable<Object>() {
+      @Override
+      public Object call() throws Exception {
```

Comment:
FYI, if you know of an efficient way of boxing []byte as []Byte, let me know ... otherwise, I would prefer to use Object here